datahub-web-react/src/app/policy/ClickOutside.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useRef } from 'react';
+import { Modal } from 'antd';
+
+interface OutsideAlerterType {
+    children: React.ReactNode;
+    onClickOutside: () => void;
+    wrapperClassName?: string;
+}
+
+export default function ClickOutside({ children, onClickOutside, wrapperClassName }: OutsideAlerterType) {
+    const wrapperRef = useRef<HTMLDivElement>(null);
+
+    const modalClosePopup = () => {
+        Modal.confirm({
+            title: 'Exit Policy Editor',
+            content: `Are you sure you want to exit policy editor? All changes will be lost`,
+            onOk() {
+                onClickOutside();
+            },
+            onCancel() {},
+            okText: 'Yes',
+            maskClosable: true,
+            closable: true,
+        });
+    };
+
+    function handleClickOutside(event) {
+        if (wrapperClassName) {
+            if (event.target && event.target.classList.contains(wrapperClassName)) {
+                modalClosePopup();
+            }
```

Comment:
this is still going to be very specific to your use case right? If we try to use `ClickOutside` anywhere else, it will open up a confirmation modal asking if we want to exit the policy editor.

I would move this `Modal.confirm(...` outside of this and pass that in as your `onClickOutside`.

Reply:
yeah, we can pass it as in `onClickOutside()`.  I will make the change and update it.